test/codeu/chat/server/ConversationAccessServerTest.java

```diff
@@ -0,0 +1,535 @@
+        "Check that the allowances can be updated",
+        controller.getPermissionMap(conversation).size()== 1);
+    
+  }
+  
+  @Test
+  public void secureCreatorByteTest(){
+	ConversationHeader conversation = spawnTestConversation();
+    HashMap<Uuid, Byte> accessMap = controller.getPermissionMap(conversation);
+  
+    assertTrue(
+        "Check that the creator's byte is 111",
+        accessMap.get(creator.id).byteValue()==creatorByte);
+    
+    assertTrue(
+        "Check that creator can access chat",
+        controller.checkMembership(creator.id, conversation));
+  }
+  
+  @Test
+  public void secureOwnerByteTest(){
+	ConversationHeader conversation = spawnTestConversation();
+	final User owner = spawnOwner(conversation);	
+	
+	assertTrue(
+	    "Check that owners can access chat",
+	    controller.checkMembership(owner.id, conversation));  
+  }
+  
+  @Test
+  public void secureMemberByteTest(){
+	ConversationHeader conversation = spawnTestConversation();
+	final User member = controller.newUser("member");
+	assertTrue(
+		"Check automatic membership for newcomers to chat",
+		controller.checkMembership(member.id, conversation));
+		    	
+	assertTrue(
+		"Check that members can access chat",
+		controller.checkMembership(member.id, conversation));
+		    
+	accessMap = controller.getPermissionMap(conversation);
+	assertFalse(
+		"Check that newcomer byte is 001",
+		accessMap.get(member.id).byteValue()==memberByte);
+  }
+  
+  @Test
+  public void secureTrollByteTest(){
+	ConversationHeader conversation = spawnTestConversation();
+	final User troll = spawnTroll(conversation);
+    
+    assertFalse(
+        "Check that trolls cannot access chat",
+        controller.checkMembership(troll.id, conversation));
+  }
+  
+  @Test
+  public void noIllegalOwnersAddedTest(){
+	ConversationHeader conversation = spawnTestConversation();
+	Uuid[] users = spawnRest(conversation);
+	  
+	controller.addOwner(creatorByte, users[0], conversation);
+	controller.addOwner(creatorByte, conversation.owner, conversation);
+	controller.addOwner(ownerByte, users[1], conversation);
+	controller.addOwner(memberByte,users[1], conversation);
+	controller.addOwner(trollByte, users[1], conversation);
+	  
+	assertTrue(
+		"Check that illegal cases of addOwner do nothing",
+		accessMap.equals(controller.getPermissionMap(conversation)));  
+  }
+  
+  @Test
+  public void properDemotionTest(){
+	ConversationHeader conversation = spawnTestConversation();
+	Uuid[] users = spawnRest(conversation);
+		
+	controller.demoteOwner(creatorByte, conversation.owner, conversation);
+	controller.demoteOwner(creatorByte, users[1], conversation);
+	controller.demoteOwner(memberByte,users[0], conversation);
+	controller.demoteOwner(ownerByte, users[0], conversation);
+	controller.demoteOwner(trollByte, users[0], conversation);
+	  
+	assertTrue(
+		"Check that illegal cases of demoteOwner do nothing",
+		accessMap.equals(controller.getPermissionMap(conversation)));
```

Comment:
This variable is initialized below this line - please fix, otherwise this won't compile.